src/main/java/uk/gov/hmcts/darts/transcriptions/service/impl/AdminTranscriptionServiceImpl.java

```diff
@@ -209,4 +214,32 @@ public TranscriptionDocumentHideResponse hideOrShowTranscriptionDocumentById(Int
 
         return response;
     }
+
+    @Transactional
+    public AdminApproveDeletionResponse approveDeletionOfTranscriptionDocumentById(Integer transcriptionDocumentId) {
+
+        transcriptionApproveMarkForDeletionValidator.validate(transcriptionDocumentId);
+
+        AdminApproveDeletionResponse response;
+
+        Optional<TranscriptionDocumentEntity> transcriptionDocumentEntity = transcriptionDocumentRepository.findById(transcriptionDocumentId);
+        if (transcriptionDocumentEntity.isPresent()) {
```

Comment:
Would be better the inverse to avoid nesting so

if (transcriptionDocumentEntity.isEmpty()) {
  throw new Exception();
}
ObjectAdminActionEntity objectAdminActionEntity = objectAdminActionRepository

Reply:
Done